lang/shared/src/main/scala/com/wavesplatform/lang/v1/compiler/CompilerV1.scala

```diff
@@ -192,7 +191,7 @@ object CompilerV1 {
       for {
         resolvedTypeParams <- TypeInferrer(typePairs).leftMap(Generic(start, end, _))
         resolvedResultType <- TypeInferrer.inferResultType(f.result, resolvedTypeParams).leftMap(Generic(start, end, _))
-        header = FunctionHeader(funcName, f.args.map(FunctionHeaderType.fromTypePlaceholder))
+        header = FunctionHeader(f.name)
```

Comment:
are you sure? why not header = f.header `or `header = FunctionHeader(f.innerName)`?
if that's an error, please write a test